osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs

```diff
@@ -60,13 +62,15 @@ public async Task StoreBeatmapSetAsync(uint beatmapSetId, byte[] beatmapPackage,
 
             await Task.WhenAll([
                 uploadBeatmapPackage(beatmapSetId, beatmapPackage, stream),
-                ..uploadAllVersionedFiles(allFiles),
-                ..uploadAllBeatmapFiles(beatmapFiles)
+                ..uploadAllVersionedFiles(beatmapSetId, allFiles),
```

Comment:
I know I already let this syntax in (by oversight) but I am not sure I can get used to this. It's quite non-obvious what this is doing.

```diff
diff --git a/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs b/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs
index f78441b..cbce08e 100644
--- a/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs
+++ b/osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs
@@ -60,11 +60,10 @@ namespace osu.Server.BeatmapSubmission.Services
                     beatmapFiles.Add((beatmapId, contents));
             }
 
-            await Task.WhenAll([
-                uploadBeatmapPackage(beatmapSetId, beatmapPackage, stream),
-                ..uploadAllVersionedFiles(beatmapSetId, allFiles),
-                ..uploadAllBeatmapFiles(beatmapSetId, beatmapFiles)
-            ]);
+            await uploadBeatmapPackage(beatmapSetId, beatmapPackage, stream);
+            await Task.WhenAll(uploadAllVersionedFiles(beatmapSetId, allFiles));
+            await Task.WhenAll(uploadAllBeatmapFiles(beatmapSetId, beatmapFiles));
+
             logger.LogInformation("All file uploads for beatmapset {beatmapSetId} concluded successfully.", beatmapSetId);
         }
 
@@ -165,7 +164,8 @@ namespace osu.Server.BeatmapSubmission.Services
             {
                 foreach (var file in files)
                 {
-                    using var response = await client.GetObjectAsync(AppSettings.S3CentralBucketName, getPathToVersionedFile(BitConverter.ToString(file.File.sha2_hash).Replace("-", string.Empty).ToLowerInvariant()));
+                    using var response = await client.GetObjectAsync(AppSettings.S3CentralBucketName,
+                        getPathToVersionedFile(BitConverter.ToString(file.File.sha2_hash).Replace("-", string.Empty).ToLowerInvariant()));
                     zipWriter.Write(file.VersionFile.filename, response.ResponseStream);
                 }
             }

```

is probably what i'd stick with.

Reply:
I'd still just use multiple awaits for better readability, but I'm not going to get hung up on it.

Reply:
I sort of explicitly didn't want to do multiple awaits, as that reduces concurrency :shrug:

Reply:
Fair, I wasn't considering that part.